src/gtc/passes/gtir_legacy_extents.py

```diff
@@ -39,13 +34,25 @@ class AssignContext:
     class StencilContext:
         assign_conditions: Dict[int, List[gtir.FieldAccess]] = field(default_factory=dict)
 
-    def visit_Stencil(self, node: gtir.Stencil, **kwargs: Any) -> FIELD_EXT_T:
-        field_extents = {name: Extent.zeros() for name in _iter_field_names(node)}
+    def visit_Stencil(
+        self, node: gtir.Stencil, *, mask_inwards: bool, **kwargs: Any
+    ) -> FIELD_EXT_T:
+        field_extents: FIELD_EXT_T = {}
         ctx = self.StencilContext()
         for field_if in node.iter_tree().if_isinstance(gtir.FieldIfStmt):
             self.visit(field_if, ctx=ctx)
         for assign in reversed(_iter_assigns(node).to_list()):
             self.visit(assign, ctx=ctx, field_extents=field_extents)
+        for name in _iter_field_names(node):
+            # ensure we have an extent for all fields. note that we do not initialize to zero in the beginning as this
+            #  breaks inward pointing extends (i.e. negative boundaries).
+            if name not in field_extents:
+                field_extents[name] = Extent.zeros()
```

Comment:
Just a comment here, there is an alternative way to write this:
```suggestion
            field_extents.setdefault(name, Extent.zeros())
```